Soma-Technologies-Inc/visitAfrica-backend
# #175494074 ft:Users should be able to make multi city request

[start 175494074]

#### What does this PR do?
- the requester should be able to make one request that has multiple destinations embedded.
- This request should then be sent to the requester's manager for approval.
- create multi city trip
#### Description of Task to be completed?
I have this endpoint working
`POST /api/trip/multi-city`
#### How should this be manually tested?
- open your terminal and clone this repo by running `git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend` 
- run `git checkout ft-multicityrequest-175494074` to navigate to my branch
- run `git pull` to make sure you have updated code
- run `npm install` to install all needed packages
- run `npm run migrate` to create migrations for the database
- navigate to login route `POST /api/auth/login` and login to get token that will b eused as a header for multi city request
- use these body data `{
    "email": "loua@gmail.com",
    "password": "Loua@123"
}`
- you will get a jwt then
- navigate to multi city request route `POST /api/trip/multi-city` and use a body data like this `{
   "departureFrom": "Romania",
   "departureTo": "Budapest",
   "startingDate": "2020-12-20",
   "returningDate": "2020-12-30",
   "multiCity": [
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      },
      {
         "name": "Budapest",
         "startingDate": "2020-12-20",
         "returningDate": "2020-12-30"
      }
   ],
   "reason": "travel"
}`
- before making a request append a `authorisation` header which have value `bearer token` replace token with the jwt token you got from `login` route
- then if you did everything correctly you will see output similar to screeenshots which are below 
#### Screenshots

![Screenshot at 2020-12-19 00-07-15](https://user-images.githubusercontent.com/64875096/102665969-df565480-4153-11eb-878e-b31674b5b126.png)
![Screenshot at 2020-12-19 00-07-45](https://user-images.githubusercontent.com/64875096/102665972-e1201800-4153-11eb-9348-9d89d154bb1f.png)
![Screenshot at 2020-12-19 00-08-08](https://user-images.githubusercontent.com/64875096/102665973-e2514500-4153-11eb-8ba0-305b65562aa3.png)


#### Questions:
#### Relevant Pivot tracker Id
[#175494074](https://www.pivotaltracker.com/story/show/175494074)
